app/gui/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -536,30 +564,25 @@ impl Container {
         let selection = Animation::new(network);
 
         frp::extend! { network
-            eval  frp.set_visibility    ((v) model.set_visibility(*v));
-            eval_ frp.toggle_visibility (model.toggle_visibility());
-
-            visualisation_uninitialised <- frp.set_visualization.map(|t| t.is_none());
-            default_visualisation <- visualisation_uninitialised.on_true().map(|_| {
+            eval input.set_view_state((state) model.apply_view_state(*state));
+            output.view_state <+ input.set_view_state.on_change();
+            output.fullscreen <+ output.view_state.map(|state| state.is_fullscreen()).on_change();
+            output.visible <+ output.view_state.map(|state| state.visible()).on_change();
+            output.size <+ input.set_size.on_change();
+
+            visualisation_uninitialised <- input.set_visualization.map(|t| t.is_none());
+            input_type_uninitialised <- input.set_vis_input_type.is_some().not();
```

Comment:
The name "visualization_uninitialized" is a bit unfortunate. I think "non-chosen" says better what it is.

Reply:
I've updated the naming.